exercises/codec-server/README.md

```diff
@@ -0,0 +1,119 @@
+   the top of `codec-server/main.go` to import the rest of your application as a
+   Go module named `temporalconverters`. Don't forget that you can import Go
+   modules from GitHub URLs.
+3. Next, create an array named `codecs` in the `main()` function of
+   `codec-server/main.go`. Keys should be Namespace strings. Values should be of
+   the type `converter.PayloadCodec`. By default, you only need to assign the
+   `default` namespace to `{temporalconverters.NewPayloadCodec()}` from this
+   example.
+4. After making these additions, you should have a functioning Codec Server,
+   integrated with your application logic. Again, everything else in here is
+   configured as generically as possible — note that this example Codec Server
+   listens on port 8081, which is usually used in testing configurations — but
+   this fulfills all the requirements of a Temporal Codec Server, and you could
+   incorporate any other authentication requirements on top of HTTP as needed.
+   Run your Codec Server with `go run ./codec-server` from the root of your
+   project directory. This will block the terminal it runs in, and await
+   connections.
+5. Now you can retrieve the decoded output of your Workflow Execution from the
+   previous Exercise. From another terminal window, run `temporal workflow show \
+   -w converters_workflowID --codec-endpoint 'http://localhost:8081/{namespace}'`.
+   It should retain the same Event History as before, with the decoded result
+   appended to the output:
+
+   ```
+   ...
+   Result:
+     Status: COMPLETED
+     Output: ["Received Plain text input"]
+   ```
+
+   You now have a working Codec Server implementation. In the following steps,
+   you'll learn how to integrate it more closely with a Temporal Cluster for
+   production environments.
+
+
+## Part B: Enable CORS and Configure Temporal Web UI Integration
+
+1. The next step is to enable Codec Server integration with the Temporal Web UI.
+   This isn't necessary if you don't plan to use the Web UI to view your
+   Workflow output, but it provides a stock example of how to integrate Codec
+   Server requests into a web app, and is supported by Temporal Cloud. Without
+   Codec Server integration, the Temporal Web UI cannot decode output, and
+   results are displayed encoded:
+
+   ![Encoded Workflow Output in Web UI](images/encoded-output.png)
+
+   To do this, you first need to enable
+   [CORS](https://en.wikipedia.org/wiki/Cross-origin_resource_sharing), a common
+   HTTP feature for securely making cross-domain requests.
+   `codec-server/main.go` contains a function called `newCORSHTTPHandler()`
+   which will add the necessary headers to your HTTP requests to support CORS,
+   but it is not enabled by default. This example Codec Server exposes an
+   additional command line parameter, `--web`, to conditionally enable CORS.
+   Restart the Codec Server with the `--web` flag: `go run ./codec-server --web
```

Comment:
I could not get the Web UI to work with the codec server. I suspect the root cause is that this command uses port 8233 (the default for the temporal CLI), while the Web UI is running on port 8080 (the standard one we use in all of our training courses).

Reply:
This is almost certainly it (I only retested that far into the exercise locally after updating Gitpod, so the port mapping was an oversight). Should be working now.